--- shell.rb	2021-05-14 10:29:11.911078079 +0000
+++ shell-updated.rb	2021-05-14 10:35:37.659066979 +0000
@@ -157,7 +157,7 @@
           ret = run_single(line)
           # don't bother saving lines that couldn't be found as a
           # command, create the file if it doesn't exist, don't save dupes
-          if ret && self.histfile && line && line.strip != @last_line&.strip
+          if self.histfile && line && line.strip != @last_line&.strip
             File.open(self.histfile, "a+") { |f| f.puts(line.strip) unless line.starts_with?(" ") }
             @last_line = line
           end
